File: src/com/leetcode/Easy/FindCommonCharacters.java
package com.leetcode.Easy;

import java.util.*;

// 1002. Find Common Characters
public class FindCommonCharacters {

    public List<String> commonChars(String[] A) {
        List<String> list = new ArrayList<>();

        int[] dict = new int[26];
        for (int i = 0; i < A[0].length(); i++) {
            dict[A[0].charAt(i) - 'a']++;
        }

        for (int i = 1; i < A.length; i++) {
            int[] curr = new int[26];

            for (int j = 0; j < A[i].length(); j++) {
                curr[A[i].charAt(j) - 'a']++;
            }

            for (int j = 0; j < 26; j++) {
                if (curr[j] < dict[j]) dict[j] = curr[j];
            }
        }

        for (int i = 0; i < 26; i++) {
            for (int j = 0; j < dict.length; j++) {
                list.add(Character.toString((char) ('a' + i)));
            }
        }

        return list;
    }
}

File: JavaScript/Easy/FindCommonCharacters.js
/**
 * 1002. Find Common Characters
 * @param {string[]} words
 * @return {string[]}
 */
const commonChars = function (words) {
  let list = [];

  let dict = new Array(26).fill(0);
  for (let i = 0; i < words[0].length; i++) {
    dict[words[0].charCodeAt(i) - 'a'.charCodeAt(0)]++;
  }

  for (let i = 1; i < words.length; i++) {
    let curr = new Array(26).fill(0);

    for (let j = 0; j < words[i].length; j++) {
      curr[words[i].charCodeAt(j) - 'a'.charCodeAt(0)]++;
    }

    for (let j = 0; j < 26; j++) {
      if (curr[j] < dict[j]) dict[j] = curr[j];
    }
  }

  for (let i = 0; i < 26; i++) {
    for (let j = 0; j < dict[i]; j++) {
      list.push(String.fromCharCode('a'.charCodeAt(0) + i));
    }
  }

  return list;
};